superset/views/core.py

```diff
@@ -161,10 +168,29 @@ class DashboardFilter(SupersetFilter):
     """List dashboards for which users have access to at least one slice"""
 
     def apply(self, query, func):  # noqa
+        if self.has_all_datasource_access() and self.has_all_dashboard_access():
+            return query
+
+        Dash = models.Dashboard  # noqa
+
+        # get ids for dashboards this user has access to
+        dashboard_perms = \
+            {d: True for d in self.get_view_menus('dashboard_access')}
+
+        dashboard_ids = []
+        for dash in db.session.query(Dash):
```

Comment:
There's concern in how getting the list of dashboards a user has access to is going to be an expensive operation. 

Ideally here we want to really limit the number of round trips to the database (it should be one complex query really, if possible), as well as the complexity of the query (if the user has all_dashboard_access, the query shouldn't do complex join to the rbac tables)

Reply:
Yes, I was concerned about that as well. But listing dashboards with this change runs at ~250ms for us, with several hundred of dashboards.  I can look at limiting the round trips though.